plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaSecretManager.java

```diff
@@ -68,7 +80,7 @@ private void waitForSecret() {
                     agentSecret = receivedValue;
                     if (connectionSecret.equals(agentSecret)) {
                         consumer.close();
-                        return;
+                        return true;
                     } else {
                         LOGGER.warning("Please sent a valid secret");
                     }
```

Comment:
Maybe `Thread.sleep()` should be added below (e.g. 100 ms or so). There is no much sense to loop non-stop here